cps/main.c

```diff
@@ -485,15 +485,19 @@ cps_proc(void *arg)
 		/*
 		 * Read in IPv4 BGP packets that arrive directly
 		 * on the Gatekeeper interfaces.
+		 *
+		 * Only needs to be called if both ntuple filter
+		 * is enabled (to steer CPS packets here) and if
+		 * RSS is enabled (to steer non-CPS packets elsewhere).
 		 */
-		if (front_iface->hw_filter_ntuple) {
+		if (front_iface->hw_filter_ntuple && front_iface->rss) {
```

Comment:
The following function should avoid repeating `front_iface->hw_filter_ntuple && front_iface->rss` so many times.
```
static inline hw_filter_ntuple_available(const struct gatekeeper_if *iface)
{
	return iface->hw_filter_ntuple && front_iface->rss;
}
```